posts/abelian-categories.md

```diff
@@ -0,0 +1,311 @@
+to the category of pointed sets. We get some useful properties: a morphism is mono if and only if it is injective
+on pseudoelements, epi if and only if it is surjective on pseudoelements, zero if and only if it maps all pseudoelements
+to zero, and a sequence is exact if and only if it is exact on pseudoelements.
+
+This is enough for some simple elementwise proofs in abelian categories, and in those cases where it works, the resulting
+proofs are just as simple as those in the category of abelian groups. However, there are major limitations:
+the functor is neither full nor faithful, so there is no "funext", and it is not possible to define morphisms based on their
+action on pseudoelements. Furthermore, the quotienting process loses the abelian group structure on morphisms, so it
+is not possible to add or subtract pseudoelements. Finally, the concept does not interact well with limits and colimits[^1].
+
+We can overcome some of the limitations of pseudoelements by not passing to a quotient and manually performing the bookkeeping
+instead. This leads to the idea of arguing "up to refinements" which was first described by George Bergman in 1974 and rediscovered
+by Joël Riou in the context of mathlib in 2023.
+
+This approach does not actually give a meaning to the term "element". Rather, it can be understood as a heuristic for translating
+proofs in the category of abelian groups to proofs in an abelian category, but the resulting proof will look a bit different
+and be a bit more complicated. This is a good fit for mathlib's abelian category and homological algebra sections, where we
+can assume that users are willing to engage with category theory, but the need remains for a tool that can be used when you just want
+your element-chasing proofs to generalize without having to think about it any further.
+
+## A project
+
+The Freyd-Mitchell embedding theorem states that for an abelian category $C$ there is a (not necessarily commutative) ring $R$ and
+a full, faithful and exact functor from $C$ to the category of $R$-modules. It was published in 1964 and provides a much stronger
+notion of pseudoelement than the one described above: our sets of pseudoelements are now modules, so they can be added and subtracted,
+we get function extensionality and we can construct morphisms by constructing $R$-linear maps of pseudoelements.
+
+This provides a satisfactory justification for chasing elements in abelian categories. The only problem is that this theorem is much
+more difficult to prove than the two techniques described above. It seemed very far out of reach when pseudoelements were added
+to mathlib in 2020. It still seemed out of reach in early 2022, but it seemed like a nice long-term goal to work towards, so
+in January 2022 I (Markus Himmel) took stock of what it would take to put a proof of the embedding theorem into mathlib.
```

Comment:
```suggestion
in January 2022 Markus Himmel took stock of what it would take to put a proof of the embedding theorem into mathlib.
```
to match with the rest of the post so far